### PR TITLE
Update to Circle 2.0

### PR DESCRIPTION
**Pull request by tysongach**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - restore_cache:
+          keys:
+            - middleman-template-{{ checksum "Gemfile.lock" }}
+
+      - run:
+          name: Install Ruby dependencies
+          command: bundle install --path vendor/bundle
+
+      - save_cache:
+          key: middleman-template-{{ checksum "Gemfile.lock" }}
```
Comment (dbernheisel) [inline .circleci/config.yml]: They recommend keying from the architecture that circle happens to use for the run. I've never seen it anything other than some linux machine, but it doesn't hurt.

something like:
`key: middleman-template-{{ arch }}-{{ checksum "Gemfile.lock" }}`

[their example](https://circleci.com/docs/2.0/caching/#full-example-of-saving-and-restoring-cache)

**Comment by dbernheisel** [comment]
LGTM. Just the one comment about specifying architecture. 👍

File: .circleci/config.yml (hunk 1 above)
Comment (tysongach) [reply .circleci/config.yml]: Oh, nice! Do you think we should also use the `{{ .Branch }}` key, and/or use multiple caches, as they do in that example? Or is that overkill in this case?

Comment (dbernheisel) [reply .circleci/config.yml]: I figure if the `Gemfile.lock` is the same across branches, it'd be ok to share the cache, so I suspect it's overkill.

**Pull request merged**